dali/kernels/imgproc/convolution/cutlass/device/gemm.h

```diff
@@ -155,72 +159,97 @@ namespace device {
       /// Number of stages used in the pipelined mainloop
       int Stages
     >
-    class Gemm;
+    class Conv;
 */
 template <
-    /// Element type for A matrix operand
-    typename ElementA_,
-    /// Layout type for A matrix operand
-    typename LayoutA_,
-    /// Element type for B matrix operand
-    typename ElementB_,
-    /// Layout type for B matrix operand
-    typename LayoutB_,
+    /// Element type for input matrix operand
+    typename ElementIn_,
+    /// Element type for input matrix operand
+    typename ElementCastIn_,
+    /// Layout type for input matrix operand
+    typename LayoutIn_,
+    /// Element type for window operands
+    typename ElementWindow_,
+    /// Element type for window operands
+    typename ElementCastWindow_,
     /// Element type for C and D matrix operands
-    typename ElementC_,
+    typename ElementOut_,
     /// Layout type for C and D matrix operands
-    typename LayoutC_,
+    typename LayoutOut_,
+    /// Type of convolution
+    bool InnerConv = true,
+    /// Convolution window storage configuration
+    typename ConvWindowConfiguration_ = ConvWindowConfiguration<1024, InnerConv>,
     /// Element type for internal accumulation
-    typename ElementAccumulator_ = ElementC_,
+    typename ElementAccumulator_ = ElementOut_,
     /// Operator class tag
     typename OperatorClass_ = arch::OpClassSimt,
     /// Tag indicating architecture to tune for
     typename ArchTag_ = arch::Sm70,
```

Comment:
Mhh. Can we have that compilation time dependent?

Reply:
I would go through the code and check if there any any other places where we need to adjust the ArchTag. Please keep in mind we support sm35 and newer.

Reply:
It's tuning, and I think for SIMT version it doesn't do anything fancy. I run it with no problem on sm61 when this tag is sm70.